### PR TITLE
feat: Container Image Exporter Plugin

### LINKED ISSUE #4609: Feature: Create a plugin to export docker image as tar file on blob storage
### 🔖 Feature description

Create a plugin to export docker image as tar file on blob storage

### 🎤 Pitch / Usecases

We want to deploy images on on-prem servers so we required docker images tar so that we can use the same image built on devtron

### 🔄️ Alternative

_No response_

### 👀 Have you spent some time to check if this issue has been raised before?

- [X] I checked and didn't find similar issue

### 🏢 Have you read the Code of Conduct?

- [X] I have read the [Code of Conduct](https://github.com/devtron-labs/devtron/blob/main/CODE_OF_CONDUCT.md)

AB#8290

### PR DESCRIPTION
**Pull request by kamal-devtron**
# Description

A plugin to export docker image as tarball

Fixes #4609 

## How Has This Been Tested?

- [x] Tested on separate devtron setup

## Checklist:

* [x] The title of the PR states what changed and the related issues number (used for the release note).
* [ ] Does this PR requires documentation updates?
* [ ] I've updated documentation as required by this PR.
* [x] I have performed a self-review of my own code.
* [x] I have commented my code, particularly in hard-to-understand areas.
* [x] I have tested it for all user roles.
* [ ] I have added all the required unit/api test cases.

### CODE REVIEW

File: scripts/sql/215_image_exporter_plugin.down.sql [hunk 1]
```diff
@@ -0,0 +1,5 @@
+DELETE FROM plugin_step_variable WHERE plugin_step_id =(SELECT ps.id FROM plugin_metadata p inner JOIN plugin_step ps on ps.plugin_id=p.id WHERE p.name='Container Image Exporter v1.0.0' and ps."index"=1 and ps.deleted=false);
+DELETE from plugin_pipeline_script where id = (SELECT script_id from plugin_step WHERE plugin_id=(SELECT id FROM plugin_metadata WHERE name='Jenkins'));
```
Comment (pawan-mehta-dt) [inline scripts/sql/215_image_exporter_plugin.down.sql:2]: Fix plugin name

File: scripts/sql/215_image_exporter_plugin.up.sql [hunk 2]
```diff
@@ -0,0 +1,82 @@
+-- Plugin metadata
+INSERT INTO "plugin_metadata" ("id", "name", "description","type","icon","deleted", "created_on", "created_by", "updated_on", "updated_by") VALUES (nextval('id_seq_plugin_metadata'), 'Container Image Exporter v1.0.0','Create Tarball of your Docker images','PRESET','https://raw.githubusercontent.com/devtron-labs/devtron/main/assets/vClutser.png','f', 'now()', 1, 'now()', 1);
```
Comment (pawan-mehta-dt) [inline scripts/sql/215_image_exporter_plugin.up.sql:2]: Improve description of plugin

File: scripts/sql/215_image_exporter_plugin.up.sql [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+    'now()',
+    1,
+    'now()',
+    1
+);
+
+
+--Plugin Step
+
+INSERT INTO "plugin_step" ("id", "plugin_id","name","description","index","step_type","script_id","deleted", "created_on", "created_by", "updated_on", "updated_by") VALUES (nextval('id_seq_plugin_step'), (SELECT id FROM plugin_metadata WHERE name='Container Image Exporter v1.0.0'),'Step 1','Creating Image Tar of image','1','INLINE',(SELECT last_value FROM id_seq_plugin_pipeline_script),'f','now()', 1, 'now()', 1);
```
Comment (pawan-mehta-dt) [inline scripts/sql/215_image_exporter_plugin.up.sql:66]: Fix description

File: scripts/sql/215_image_exporter_plugin.up.sql [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+
+
+-- Input Variables
+
+INSERT INTO "plugin_step_variable" ("id", "plugin_step_id", "name", "format", "description", "is_exposed", "allow_empty_value", "variable_type", "value_type", "variable_step_index", "deleted", "created_on", "created_by", "updated_on", "updated_by") VALUES
+(nextval('id_seq_plugin_step_variable'), (SELECT ps.id FROM plugin_metadata p inner JOIN plugin_step ps on ps.plugin_id=p.id WHERE p.name='Container Image Exporter v1.0.0' and ps."index"=1 and ps.deleted=false), 'Platform','STRING','Please specify the platform architecture of the image being exported (arm64 or amd64).',true,false,'INPUT','NEW',1 ,'f','now()', 1, 'now()', 1),
```
Comment (pawan-mehta-dt) [inline scripts/sql/215_image_exporter_plugin.up.sql:72]: Remove please statements from variable descriptions

File: scripts/sql/215_image_exporter_plugin.up.sql [hunk 5]
```diff
@@ -0,0 +1,82 @@
...
+(nextval('id_seq_plugin_step_variable'), (SELECT ps.id FROM plugin_metadata p inner JOIN plugin_step ps on ps.plugin_id=p.id WHERE p.name='Container Image Exporter v1.0.0' and ps."index"=1 and ps.deleted=false), 'AzureAccountName','STRING',' Specify the name of your Azure storage account.',true,true,'INPUT','NEW',1 ,'f','now()', 1, 'now()', 1),
+(nextval('id_seq_plugin_step_variable'), (SELECT ps.id FROM plugin_metadata p inner JOIN plugin_step ps on ps.plugin_id=p.id WHERE p.name='Container Image Exporter v1.0.0' and ps."index"=1 and ps.deleted=false), 'FilePrefix','STRING','If youd like to add a prefix to the exported image files name,  enter it here.',true,false,'INPUT','NEW',1 ,'f','now()', 1, 'now()', 1),
+(nextval('id_seq_plugin_step_variable'), (SELECT ps.id FROM plugin_metadata p inner JOIN plugin_step ps on ps.plugin_id=p.id WHERE p.name='Container Image Exporter v1.0.0' and ps."index"=1 and ps.deleted=false), 'CloudProvider','STRING',' Provide which cloud storage provider you want to use: "aws" for Amazon S3 or "azure" for Azure Blob Storage.',true,false,'INPUT','NEW',1 ,'f','now()', 1, 'now()', 1),
```
Comment (pawan-mehta-dt) [inline scripts/sql/215_image_exporter_plugin.up.sql:79]: Convert all input values to lower case so that the script doesn't fail with different inputs

**Review by pawan-mehta-dt — APPROVED** [review]

**Review by kripanshdevtron — APPROVED** [review]

**Pull request merged**